src/Runner.Worker/Handlers/StepHost.cs

```diff
@@ -111,7 +111,10 @@ public string ResolvePathForStepHost(string path)
         {
             // make sure container exist.
             ArgUtil.NotNull(Container, nameof(Container));
-            ArgUtil.NotNullOrEmpty(Container.ContainerId, nameof(Container.ContainerId));
+            if (!FeatureFlagManager.IsHookFeatureEnabled())
```

Comment:
We should check here if the hook feature is enabled and the user has opted into using it by setting the env variable.